### DISCUSSION TITLE
Unsupervised Shapelets

### DISCUSSION THREAD

**Discussion by Corkman99** [tslearn-team/tslearn#523] [General]
Hi! 

I am looking at the LearningShapelets module for a time-series regression task and would like to perform unsupervised learning. The available documentation includes a function of the LearningShapelets class called fit_transform(X,y,...) where if y=None, then the function learns Shapelets without labels. In the Repo I cannot find this function, and the fit() function requires labels. Did you get rid of this feature? Do you have any tips and existing code that could help me? 

Thanks! 
![Screenshot 2024-07-26 101901](https://github.com/user-attachments/assets/9f21d53b-f2fd-4ac6-87a9-26c6f5ef649f)
